### DISCUSSION TITLE
Is is possible to increase the build queue size?

### DISCUSSION THREAD

**Discussion by infomiho** [caprover/caprover#1939] [Q&A] [answered]
Hey 👋

I've been enjoying transferring my apps to CapRover. My app setup includes:
- server
- client
- DB

and I keep it in a monorepo (https://github.com/infomiho/socialpostgpt).

I've managed to get it to automatically deploy via Github Webhooks which is great! (https://socialpostgpt.xyz/)

**But, when I push some changes there is a delay between deploying the `client` and the `server` apps due to one being queued while the other is being built. Is there a way to increase the queue size?**

Off topic:
Ideally, the apps `client` and `server` would be connected somehow and  their images deployed at the same time. But that's probably a feature request.

**Comment by githubsaturn** [comment, marked as answer]
Short answer: No.

A very longer answer:
- Creating a PR to add the queue size to be configurable is welcomed!
- You should definitely look into [building the image on github](https://github.com/caprover/deploy-from-github#example-1---deploy-using-image), and not use your server's CPU/RAM for build. 
- The change in the build queue most definitely doesn't solve your problem as best it reduces the negative impact. What if there is another app that is being deployed at the same time? What if you deploy two versions of client/server back to back? What about the fact that your client build time will be definitely different from the server code build time... 
- To do this properly, you have some options: method 1: deploy the additional functionality in the backend first, and then only after you confirmed that the API works fine, you deploy the frontend. method 2: deploy at the same time, but have login in the frontend to check the backend capabilities before showing the changes. method 3: use a single binary! Your frontend code is simply some html / static files, you can definitely serve them directly from your backend code


>  Ideally, the apps client and server would be connected somehow and their images deployed at the same time. But that's probably a feature request.

"Deploying at the same time" in distributed systems is not possible. Deployment is a async process. Unless you bundle your client/server in the same binary, it's not possible.

**Reply by infomiho** [reply]
Thank you for the quick answer!

> Creating a PR to add the queue size to be configurable is welcomed!

I'm keen on contributing to Caprover since I really love the self hosted PaaS idea and the codebase is in my language of choice. I'll find time to go through the repo and create a PR 🤞

> You should definitely look into [building the image on github](https://github.com/caprover/deploy-from-github#example-1---deploy-using-image), and not use your server's CPU/RAM for build.

I've migrated to using Github Actions as my CI and deploying the server and client apps via the built images. This reduces the time between server and client deployment which is acceptable for my kind of side projects :) 

Here's an example run: https://github.com/infomiho/phrase-tutor-wasp/actions/runs/6840817663

> "Deploying at the same time" in distributed systems is not possible. Deployment is a async process. Unless you bundle your client/server in the same binary, it's not possible.

This is great reminder for me. That's the reality of multi service systems. Thank you for the detailed breakdown 👍  The option to bundle both server and the client is something we at [Wasp](https://github.com/wasp-lang/wasp) are looking to support in the next quarter. It will be a configuration option: multiple container or a single container with the full-stack app 🙂
